Core/Src/main.c

```diff
@@ -582,12 +582,40 @@ void HAL_DAC_ConvCpltCallbackCh1(DAC_HandleTypeDef *hdac) {
 	HAL_DAC_Stop_DMA(&hdac1, DAC_CHANNEL_1);
 	if (toneIndex >= toneSequenceSize) {
 		playingSequence = 0;
+		toneIndex = 0;
+		HAL_DFSDM_FilterRegularStart_DMA(&hdfsdm1_filter0, currentTone, TONE_LEN);

```

Comment:
This is interesting -- part of me would be tempted to have this be separate (eg, call `playSequenceBlocking()` and then have a separate method to get mic input. My reasoning is mostly just for debugging, it would be nice to be able to just test recording and matching a tone independently of the DAC operation.

Reply:
Actually, I came up with a better reason, which is that we likely don't want to start recording tones immediately after the sequence is played, but have some sort of setup where the user presses a button to record their sequence

Reply:
That's a good reason, we can leave it like this for now but I do agree a second button click or something might be worth it so I will note to come back to this later